src/suspense.js

```diff
@@ -1,23 +1,102 @@
 import { Component } from './component';
+import { unmount } from './diff/index';
+import { removeNode } from './util';
 import { createElement } from './create-element';
 
-// having a "custom class" here saves 50bytes gzipped
-export function Suspense(props) {}
+function removeDom(vnode) {
+	let tmp = vnode._component;
+	if (tmp != null) {
+		if (tmp = tmp._prevVNode) removeDom(tmp);
+	}
+	else if ((tmp = vnode._children) && vnode._lastDomChild) {
+		for (let i = 0; i < tmp.length; i++) {
+			if (tmp[i]) removeDom(tmp[i]);
+		}
+	}
+
+	if (tmp = vnode._dom) {
+		removeNode(tmp);
+	}
+}
+
+// having custom inheritance instead of a class here saves a lot of bytes
+export function Suspense(props) {
+	// we do not call super here to golf some bytes...
+	this._suspensions = [];
+}
+
+// Things we do here to save some bytes but are not proper JS inheritance:
+// - call `new Component()` as the prototype
+// - do not set `Suspense.prototype.constructor` to `Suspense`
 Suspense.prototype = new Component();
 
 /**
  * @param {Promise} promise The thrown promise
  */
 Suspense.prototype._childDidSuspend = function(promise) {
-	this.setState({ _loading: true });
-	const cb = () => { this.setState({ _loading: false }); };
+	this._suspensions.push(promise);
+	let len = this._suspensions.length;
+
+	const suspensionsCompleted = () => {
+		// make sure we did not add any more suspensions
+		if (len === this._suspensions.length) {
+			// clear old suspensions
+			this._suspensions = [];
+
+			// reset the timeout
+			this._timeout = null;
+
+			// remove fallback
+			unmount(this.props.fallback);
+
+			// make preact think we had mounted the parkedVNode previously...
+			this._prevVNode = this._parkedVnode;
+			this._parkedVnode = null;
```

Comment:
Can we delay the this._timeout setting to here so you can do `this._timeout = this._parkedVNode = null`

Reply:
Excellent find. Will check when I‘m on my computer!

Reply:
Done. Thx!